test/gtest/ucp/test_ucp_tag.cc

```diff
@@ -490,79 +565,136 @@ class test_ucp_tag_nbx : public test_ucp_tag {
     {
-        ucp_request_param_t send_param = sparam;
-        ucp_request_param_t recv_param = rparam;
-
-        if (prereg) {
-            send_param.op_attr_mask |= UCP_OP_ATTR_FIELD_MEMH;
-            send_param.memh          = sender().mem_map(
-                                               const_cast<void*>(send_buffer),
-                                               size);
-
-            recv_param.op_attr_mask |= UCP_OP_ATTR_FIELD_MEMH;
-            recv_param.memh          = receiver().mem_map(recv_buffer, size);
-        }
+        const ssize_t recv_quantity = recv_dt.tag_quantity();
+        ucs_assert(recv_quantity > 0);
 
         ucs_status_ptr_t recv_req = ucp_tag_recv_nbx(receiver().worker(),
-                                                     recv_buffer, size, 0, 0,
-                                                     &recv_param);
+                                                     recv_dt.buf(),
+                                                     recv_quantity, 0,
+                                                     0, &rparam);
         ASSERT_UCS_PTR_OK(recv_req);
 
-        ucs_status_ptr_t send_req = ucp_tag_send_nbx(sender().ep(), send_buffer,
-                                                     size, 0, &send_param);
+        const ssize_t send_quantity = send_dt.tag_quantity();
+        ucs_assert(send_quantity > 0);
+        ucs_status_ptr_t send_req = ucp_tag_send_nbx(sender().ep(),
+                                                     send_dt.buf(),
+                                                     send_quantity, 0,
+                                                     &sparam);
         ASSERT_UCS_PTR_OK(send_req);
 
-        if (!(recv_param.op_attr_mask & UCP_OP_ATTR_FIELD_REQUEST)) {
+        if (!(rparam.op_attr_mask & UCP_OP_ATTR_FIELD_REQUEST)) {
             request_wait(recv_req);
         }
 
-        if (!(send_param.op_attr_mask & UCP_OP_ATTR_FIELD_REQUEST)) {
+        if (!(sparam.op_attr_mask & UCP_OP_ATTR_FIELD_REQUEST)) {
             request_wait(send_req);
         }
+    }
+
+    void test_prereg_rcache_stats(const ucp::data_type_desc_t &send_dt,
+                                  ucp::data_type_desc_t &recv_dt,
+                                  const ucp_request_param_t &sparam,
+                                  const ucp_request_param_t &rparam)
+    {
+        int rcache_count;
+
+        if (sparam.op_attr_mask & UCP_OP_ATTR_FIELD_REQUEST) {
+            /* Can't send multiple requests for a single user request memory
+               handle */
+            return;
+        }
+
+        for (int i = 0; i < 10; ++i) {
+            do_send_recv(send_dt, recv_dt, sparam, rparam);
+        }
+
+        rcache_count = UCS_STATS_GET_COUNTER(sender().ucph()->rcache->stats,
+                                             UCS_RCACHE_GETS);
+
+        /* mem_map uses rcache twice ('lookup + get') */
+        EXPECT_EQ(2, rcache_count);
```

Comment:
test that count before the 10-iteration loop and after it are equal?
to not assume how many times get was done before this function was called